### ISSUE TITLE
Expose new resource detector and point @opentelemetry/resource-detector-gcp at it

### ISSUE THREAD

**Issue by aabmass** [GoogleCloudPlatform/opentelemetry-operations-js#518]
The resource detector added in `@google-cloud/opentelemetry-resource-util` is not exported at this time. I need to expose and document it.

There is an existing upstream resource detector for GCP: https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/detectors/node/opentelemetry-resource-detector-gcp `@opentelemetry/resource-detector-gcp`

`@opentelemetry/resource-detector-gcp` should either
1. be deprecated
2. compose/re-export the one in this repo

**Comment by aabmass** [comment]
triaged and assigned to myself

**Comment by anuraaga** [comment]
@aabmass Was looking at resource for a NodeJS app and found these two big differences between upstream and this package

- Upstream populates more k8s information such as K8S_POD_NAME (I couldn't find it in this repo but let me know if I missed it)
- Upstream does not populate CLOUD_PLATFORM

The latter is particularly important because the resource transformation logic in this repo relies heavily on CLOUD_PLATFORM being set correctly. So for now, I am using both packages but presumably the CLOUD_PLATFORM logic needs to be upstreamed and then this repo's package could be deprecated

**Comment by aabmass** [comment]
Yes, the plan is to replace the implementation of `@opentelemetry/resource-detector-gcp` with the library provided here. I just haven't had time to do it.

> * Upstream populates more k8s information such as K8S_POD_NAME (I couldn't find it in this repo but let me know if I missed it)

@dashpole should we be detecting this from hostname like the [upstream detector does](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/82267ab02f8d9b27613b5926089c42d04d4a4a7e/detectors/node/opentelemetry-resource-detector-gcp/src/detectors/GcpDetector.ts#L83), since https://github.com/open-telemetry/oteps/pull/195 fell through the cracks?

**Comment by dashpole** [comment]
Personally, I don't think k8s_pod_name detection belongs in a GCP detector, as it isn't GCP-specific.  That being said, its fine if it stays in the GCP detector if there isn't an alternative k8s detector that detects it.

**Comment by aabmass** [comment]
I agree it would be nice to push these into generic k8s resource detectors

**Comment by anuraaga** [comment]
BTW, I noticed this in Go

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/gcp/gke.go#L31

If the generally accepted approach is to populate k8s information in the manifest through `OTEL_RESOURCE_ATTRIBUTES`, then the population of k8s information that upstream gcp detector is currently doing isn't actually needed in the long term. Either way, for my app I went with removing use of the upstream one and only use the one from this repo now as I'm setting that env variable anyways for Go apps now.

**Comment by dashpole** [comment]
xref: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4136

**Comment by aabmass** [comment]
Those PRs were merged, the upstream detector `@opentelemetry/resource-detector-gcp` should be used instead: https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/packages/resource-detector-gcp

https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/pull/840 already updated the e2e tests in this repo to use the upstream one. So the remaining work here would be to mark the one in this repo deprecated on NPM.

**Comment by aabmass** [comment]
#948 
#946 
#933
